### PR TITLE
Update external vision segment in ECL/EKF

### PR DESCRIPTION
**Pull request by kamilritz**

### CODE REVIEW

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 1]
```diff
@@ -191,11 +191,26 @@ Optical flow data will be used if the following conditions are met:
 
 ### External Vision System
 
-Position and Pose Measurements from an external vision system, e.g. Vicon, can be used:
+Position, Velocity or Orientation Measurements from an external vision system, e.g. Vicon, can be used:
```
Comment (hamishwillee) [inline en/advanced_config/tuning_the_ecl_ekf.md]: Pose is orientation. Either way, can you lower case velocity and orientation.

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 2]
```diff
@@ -191,11 +191,26 @@ Optical flow data will be used if the following conditions are met:
...
-* External vision system pose data will be used for yaw estimation if bit position 4 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision system velocity data will be used if bit position 8 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision system orientation data will be used for yaw estimation if bit position 4 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision reference frame offset will be estimated and used to rotate the external vision system data if bit position 6 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+
+Either bit 4 or bit 6 should be set to true. Following EKF_AID_MASK values are supported when using with an external vision system.
+
+EKF_AID_MASK value | Set bits | Description
+--- | --- | --- 
+73 | GPS + EV_POS + ROTATE_EV | *not recommended*, use EV_VEL instead, heading wrt north
```
Comment (hamishwillee) [inline en/advanced_config/tuning_the_ecl_ekf.md]: For all cases, please change:
- wrt to w.r.t.
- north to North

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 3]
```diff
@@ -191,11 +191,26 @@ Optical flow data will be used if the following conditions are met:
...
+321 | GPS + EV_VEL + ROTATE_EV | *recommended*, heading wrt north
```
Comment (hamishwillee) [inline en/advanced_config/tuning_the_ecl_ekf.md]: Worth moving the thing you want people to do (recommended) to the top?

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 4]
```diff
@@ -191,11 +191,26 @@ Optical flow data will be used if the following conditions are met:
...
 * External vision system vertical position data will be used if the [EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE) parameter is set to 3.
-* External vision system pose data will be used for yaw estimation if bit position 4 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision system velocity data will be used if bit position 8 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision system orientation data will be used for yaw estimation if bit position 4 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision reference frame offset will be estimated and used to rotate the external vision system data if bit position 6 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+
+Either bit 4 or bit 6 should be set to true. Following EKF_AID_MASK values are supported when using with an external vision system.
```
Comment (hamishwillee) [inline en/advanced_config/tuning_the_ecl_ekf.md]: Backticks around `EKF_AID_MASK`

**Comment by hamishwillee** [comment]
@kamilritz Looks really useful  - thanks.

@TSC21 Can you verify it is correct.

File: en/advanced_config/tuning_the_ecl_ekf.md (hunk 1 above)
Comment (kamilritz) [reply en/advanced_config/tuning_the_ecl_ekf.md]: Pose is the combination of position and orientation. https://en.wikipedia.org/wiki/Pose_(computer_vision)

**Comment by kamilritz** [comment]
@hamishwillee  Thank you for the feedback

**Comment by hamishwillee** [comment]
@TSC21 Can you please review this? Or suggest best alternative to yourself for this purpose?

**Comment by hamishwillee** [comment]
Thanks very much @kamilritz (and @TSC21 !). Much appreciated. Merged.

**Pull request merged**